test/common/upstream/health_checker_impl_test.cc

```diff
@@ -2170,65 +2211,55 @@ class TcpHealthCheckerImplTest : public testing::Test {
       : cluster_(new NiceMock<MockClusterMockPrioritySet>()),
         event_logger_(new MockHealthCheckEventLogger()) {}
 
-  void setupData() {
-    std::string json = R"EOF(
-    {
-      "type": "tcp",
-      "timeout_ms": 1000,
-      "interval_ms": 1000,
-      "unhealthy_threshold": 2,
-      "healthy_threshold": 2,
-      "send": [
-        {"binary": "01"}
-      ],
-      "receive": [
-        {"binary": "02"}
-      ]
-    }
+  void setupData(unsigned int unhealthy_threshold = 2) {
+    std::ostringstream yaml;
+    yaml << R"EOF(
+    timeout: 1s
+    interval: 1s
+    unhealthy_threshold: )EOF"
+         << unhealthy_threshold << R"EOF(
+    healthy_threshold: 2
+    tcp_health_check:
+      send:
+        text: "01"
```

Comment:
If I'm reading the proto file correctly, this is the right way to send binary encoded as hex stream (and the unit tests that use this still pass), but wouldn't mind a double-check here.

https://github.com/envoyproxy/envoy/blob/31485b5d007a87352853ca16d5ee5b91e400d85b/api/envoy/api/v2/core/health_check.proto#L80-L81